provide link to attach propales to interventions and a way to restrict selection of links to a specific thirdparty

bafbes opened the pull request.

bafbes committed:
provide link to attach propales to interventions and a way to restrict selection of links to a specific thirdparty

Conflicts:
	htdocs/core/class/html.form.class.php

eldy commented on htdocs/core/class/html.form.class.php:
```diff
@@ -5024,7 +5024,7 @@ function showLinkedObjectBlock($object)
      *  @return	int									<0 if KO, >0 if OK
      */
-    function showLinkToObjectBlock($object, $restrictlinksto=array())
+    function showLinkToObjectBlock($object, $restrictlinksto=array(),$fk_soc=0)
```
Adding a parameter must also be added into the code comments of function.
Also we should not need this parameter, the $object->fk_soc contains already the company

eldy commented on htdocs/comm/propal.php:
```diff
@@ -2256,6 +2256,12 @@
 		$linktoelem = $form->showLinkToObjectBlock($object);
 		if ($linktoelem) print '<br>'.$linktoelem;
 
+        // Show links to link elements
```
We should not need this. All links we want to see should be reported into showLinkToObjectBlock.
So if a special cas must be managed, it should be inside function

bafbes committed:
fix:showLinkToObjectBlock should contain one parameter only

Conflicts:
	htdocs/comm/propal.php
	htdocs/core/class/html.form.class.php
	htdocs/expensereport/card.php

bafbes committed:
conflicts solved Merge branch 'develop' into abb-4

Conflicts:
	htdocs/core/class/html.form.class.php
	htdocs/expensereport/card.php

bafbes commented:
conflicts solved

eldy commented:
Somthing seems wrong with PR submission.
817 files modified compared to develop branch !  So i Close it.

bafbes commented:
I will correct it and recreate it from scratch later.
